### ISSUE TITLE
Inconsistent Case for App Name in Discovery

### ISSUE THREAD

**Issue by eaglerainbow** [promregator/promregator#77]
## Environment
Promregator v0.5.0-rc5

## Summary / Problem Statement
The case of the application name may be different if discovered using `applicationRegex` compared `applicationName`

## Observed Behavior
Let there be an application called "testApp" in a space that we want to discover. The target configuration

```yaml
  targets:
    - orgName: org
      spaceName: space
      applicationName: testapp
```
will result in a discovery where the application is called `testapp` (like in the configuration). 

The configuration
```yaml
  targets:
    - orgName: org
      spaceName: space
      applicationRegex: testapp
```
(mind that we run over regexing now) results in an application called `testApp`. 

This is inconsistent.

## Expected Behavior
In both cases, the application shall be called `testApp`.

**Comment by eaglerainbow** [comment]
Problem is root-caused by https://github.com/promregator/promregator/blob/aebdde1b5ab366a5cc293a18857f3b00b7042546/src/main/java/org/cloudfoundry/promregator/scanner/ReactiveTargetResolver.java#L45 where we just one-to-one map the `applicationName` of the `configTarget` to the `ResolvedTarget` without having verified that this app exists.

**Comment by eaglerainbow** [comment]
Postponing to work on this issue a little: With the upcoming of `orgRegex` and `spaceRegex` the coding there has to change significantly anyway. Only afterwards we should revisit this bug here.

**Comment by eaglerainbow** [comment]
Solved with 61ddf9b02136468346338ec9bb90348a3fdd574f, which makes the `ReactiveTargetResolver` upside down to support `orgRegex`and `spaceRegex` (see also issue #35).
